test/event/handlers/CommandParserHandlerTest.ts

```diff
@@ -92,6 +93,23 @@ describe("CommandParserHandler", () => {
 			expect(runCommandMock.called).to.be.false;
 		});
 
+		it("should not throw error if message consists in command prefix", async () => {
+			const message = discordMock.getMessage();
+
+			message.content = COMMAND_PREFIX;
+			message.channel.id = "fake-bot-enabled-channel-id";
+
+			let errorWasThrown = false;
+
+			try {
+				await handler.handle(message);
+			} catch (error) {
+				errorWasThrown = true;
```

Comment:
I have to admit this is not optimal. I tried using Chai's error testing API with something along these lines:
```js
await handler.handle(message);

const error = new TypeError("Cannot read property 'toLowerCase' of undefined");

expect(CommandFactory.prototype.commandExists).not.to.throw(error);
```

But I couldn't manage to make it work. If anybody knows a nicer way of doing this I'll be happy to fix it.

Reply:
Yeah, Chai is rather weird about handling Promise rejections. I remember @Jason2605 and I trying to figure it out. I'm happy with this.